psi4/src/psi4/libfock/DirectDFJK.cc

```diff
@@ -0,0 +1,1600 @@
+			Block_funcs_.push_back(block_size);
+			block_size = shell_funcs;
+			Shell_stops_.push_back(shell_iter - 1);
+			Shell_starts_.push_back(shell_iter);
+		} else {
+			block_size += shell_funcs;
+			if (block_size > bmf) {
+				 bmf = block_size;
+			}
+		}
+	}
+	Shell_stops_.push_back(p_shells_ - 1);
+
+	Block_funcs_.push_back(block_size);
+
+	biggest_block_ = bmf * naux_ * nbf_;
+	num_blocks_ = Block_funcs_.size();
+
+    k_disps_.resize(num_blocks_);
+	size_t col_disp = 0U;
+	size_t row_disp = 0U;
+
+	for (size_t row_it = 0; row_it < num_blocks_; row_it++) {
+		k_disps_[row_it].resize(num_blocks_);
+		for (size_t col_it = 0; col_it < num_blocks_; col_it++) {
+			k_disps_[row_it][col_it] = nbf_* row_disp + col_disp;
+			col_disp += Block_funcs_[col_it];
+		}
+		row_disp += Block_funcs_[row_it];
+		col_disp = 0U;
+	}
+
+}
+
+//decides where all the blocks will stop and start
+// initializes Shell_starts_ and Shell_stops_
+// initializes num_blocks_
+void DirectDFJK::prepare_Q_blocks() {
+//	std::vector<size_t> starts;//Shell_starts_.get();
+//	std::vector<size_t> stops;//stops_.get();
+	size_t current_costs = 0;
+	double charges_f = 0.0;
+	size_t charges_z = 0;
+
+	std::shared_ptr<Molecule> mol_ptr = primary_->molecule();
+
+	for (int atom_iter = 0; atom_iter < mol_ptr->natom(); atom_iter++)
+		charges_f+=mol_ptr->fZ(atom_iter);
+                                                                   
+	charges_f = charges_f/2.0;
+                                                                   
+	charges_z = static_cast<size_t>(charges_f);
+	++charges_z;
+
+	current_costs += 3*sizeof(size_t)*nbf_*nbf_;
+	current_costs += 2*sizeof(double)*naux_*naux_;
+	current_costs += 4*sizeof(double)*nbf_*nbf_;
+	current_costs += 4*sizeof(double)*nbf_*nbf_;
+
+	size_t block_max = (free_memory_)/2;
+	size_t block_size = 0;
+	size_t shell_funcs;
+
+	Shell_starts_.push_back(0);
+
+	biggest_block_ = 0;
+
+	for( size_t shell_iter = 0; shell_iter < Q_shells_; shell_iter++) {
+		shell_funcs = auxiliary_->shell(shell_iter).nfunction();
+		if ((shell_funcs+block_size+3)*nbf_*(nbf_+2*charges_z)>memory_-current_costs) {
+			Block_funcs_.push_back(block_size);
+			block_size = shell_funcs;
+			Shell_stops_.push_back(shell_iter - 1);//[block_iter] = shell_iter - 1;
+			Shell_starts_.push_back(shell_iter);//[block_iter] = shell_iter;
+		} else {
+			block_size += shell_funcs ;
+			if ( block_size *nbf_ * nbf_> biggest_block_ ) { 
+				biggest_block_ = block_size * nbf_ * nbf_; 
+			}
+		}
+	}
+
+	Block_funcs_.push_back(block_size);
+
+	Shell_stops_.push_back(Q_shells_-1 );
+	
+	num_blocks_ = Shell_starts_.size();
+}
+
+void DirectDFJK::compute_dense_AO_block_p_pQq(size_t shell, double* ao_block, std::vector<std::shared_ptr<TwoBodyAOInt>> eri){
+    int procs = 1;
+#ifdef _OPENMP
+	procs = omp_nthread_;
+#endif
+    
+
+    std::vector<const double*> buffer(procs);
+#pragma omp parallel num_threads(procs)
```

Comment:
This should be within `#ifdef _OPENMP`

Reply:
Will change

Reply:
Actually I take that back. I've looked up the API, and your suggested change doesn't seem like it will change anything. I've doublechecked in other parts of psi, and the same syntax as I have here is used there as well.

Reply:
Well, it makes compilers warn about pragmas that have no effect.

Reply:
You're asking me to make changes in my code that will still leave users with hundreds of the same compiler warnings from other parts of the codebase. I don't see how your suggestion affects the end user.

Reply:
code quality ftw :rofl:

Reply:
If it's part of a larger change in standards in psi to eliminate these messages from the compiler, I will be happy to make the change. I can understand the wish to enforce strict standards in a package where the entire world can see you as a core dev.